### DISCUSSION TITLE
Add a resource to the `api/artifacts` endpoint to obtain versions of a given artifact

### DISCUSSION THREAD

**Discussion by jyoo980** [scalacenter/scaladex#997] [Ideas]
Following from [the discussion](https://github.com/scalacenter/scaladex/pull/992#discussion_r841492605) from #992, cc @adpi2 @mlachkar, @julienrf.

I'm happy to write up an issue/ticket for this task (and possibly take it on), but I was wondering if folks had any preferences for how the resource should look like. I was thinking something like:

`api/artifacts/<artifact name>/versions`

But I'm not sure how flexible that would be. I think it would also be a good idea to think about how the response should look like. Do we want to return just the versions by themselves, or organized in some way that makes sense, e.g., something akin to the version matrix on the index. Any thoughts?

**Comment by adpi2** [comment]
That's a good question @jyoo980.

First thing is that we need the `<group_id>` in the url to distinguish between identical artifact names in different projects: `api/artifacts/<group_id>/<artifact_name>`

For instance, the artifact name `core` is used in many different projects.

Then the question is, should we use `artifact_name` or `artifact_id`? 

`artifact_name` is more flexible because we can add the `language` and `platform` filters: one can load all the versions of an artifact that are compatible with Scala 3 and Scala.js 1.x. But then also, as you said, we should organize the versions in some way, e.g., something akin the version matrix. This would make it powerful but also harder for downstream users to consume: they need to parse the response and use some logic to build the `artifact_id` by combining `artifact_name`, `language` and `platform`. This is not impossible but also not convenient. In the long term, I am thinking of publishing a Scaladex library that can do all that: call the API, parse the response into our model and use our logic. But we are not there yet: we first need to make sure that the current model is stable.

For the time being we should go with `<artifact_id>` because it makes the endpoint simpler and more simple to use.

So the endpoint should be: 
```
/api/artifacts/<group_id>/<artifact_id>/
```
and the response should be `Page[String]` where `String` is the string representation of `SemanticVersion`.

Using `<artifact_id>` instead of `<artifact_name>` is also consistent with the response of `/api/artifacts` in #992.

**Comment by jyoo980** [comment]
Issue opened [here](https://github.com/scalacenter/scaladex/issues/998) to capture this work. Will populate it later with more information.
